src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -130,16 +139,30 @@ abstract class AbstractNormalizer implements NormalizerInterface, DenormalizerIn
 
     /**
      * @var NameConverterInterface|null
+     *
+     * @deprecated since Symfony 5.1, use $nameConverters instead.
      */
     protected $nameConverter;
 
+    /**
+     * @var array<string, NameConverterInterface>
+     */
+    protected $nameConverters = [];
+
     /**
      * Sets the {@link ClassMetadataFactoryInterface} to use.
      */
-    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, array $defaultContext = [])
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, /* array */ $nameConverter/*s*/ = [], array $defaultContext = [])
     {
         $this->classMetadataFactory = $classMetadataFactory;
-        $this->nameConverter = $nameConverter;
+        if (!\is_array($nameConverter)) {
+            @trigger_error(sprintf('The 2nd constructor argument of the class "%s" should be an array of name converters, using a name converter directly or null is deprecated since Symfony 5.1.', __CLASS__), E_USER_DEPRECATED);
+            $this->nameConverter = $nameConverter;
+        } else {
+            foreach ($nameConverter as $nameConverterInstance) {
+                $this->nameConverters[\get_class($nameConverterInstance)] = $nameConverterInstance;
```

Comment:
Is it a good idea to use the FQCN for referencing a name converter?
We could also use the service id.

Reply:
nope, it's not: it prevents passing two differently configured instances of the same class
instead, why not pass a service locator?

Reply:
We could use a service locator too.
We would keep the name converter and we would add one more argument, am I right?
Or maybe we can replace the name converter with the service locator. I'm not sure to like it since it means not having a clear dependency between the normalizer and the name converter and having to use a default name converter service id.